### PR TITLE
fixes #150 Backend modified for transaction spanning across multiple wallets

### LINKED ISSUE #150: Spend in single transaction outputs managed by multiple wallets
**Feature description**

It should be possible to spend outputs no matter where them found.

**Is your feature request related to a problem? Please describe.**
It's impossible to aggregate funds kept in multiple wallets for spending in a single transaction even if private keys are well known .

**Describe the solution you'd like**
Support for multiple address and wallet selection in advanced spend GUI

**Describe alternatives you've considered**
At present only a single wallet can be chosen .

**Additional context**
This requires support for partial signing of SkyFiber transactions, documented in #138 .

**Possible implementation**

- [x] Multiple wallet selection in advanced spend GUI
- [ ] Connect wallet selection GUI with backend via QT models
- [x] If transaction spawns across multiple wallets iterate over available strategies for partial signing
- [ ] Ensure that transaction is fully signed at the end of the process 
- [x] UXID array lookup IDs for signing support #<NUMBER> relative to transaction array
- [x] UXID array lookup IDs for signing support absolute <UXID>
- [x] Add BlockchainTransactionAPI interface for creating multi-wallet transactions
- [x] Add BlockchainSignService interface for signing multi-wallet transactions
- [x] New altcoin plugin entry points for instantiating aforementioned API's
- [x] Tests for multi-wallet interfaces
- [ ] GUI component for signing progress and password input

### PR DESCRIPTION
**Pull request by olemis**
Refs #150

Changes:
- UXID array lookup IDs for signing now support `#<NUMBER>` relative to transaction array
- UXID array lookup IDs for signing now support absolute `<UXID>`
- Add `BlockchainTransactionAPI` interface for creating multi-wallet transactions
- Add `BlockchainSignService` interface for signing multi-wallet transactions
- New altcoin plugin entry points for instantiating aforementioned API's

Does this change need to mentioned in CHANGELOG.md?
yes

Requires testing
yes

Test cases modified to test for UXID index lookup. New test cases added for muli wallet spend behavior.

### CODE REVIEW

File: src/coin/skycoin/models/wallet.go [hunk 1]
```diff
@@ -1057,7 +1094,7 @@ func (wlt *LocalWallet) signSkycoinTxn(txn core.Transaction, pwd core.PasswordRe
 			logWallet.Errorf("Error parsing fee of TxID %s : %s", cTxn.TxID, cTxn.Fee)
 			return nil, err
 		}
-		txnFee = uint64(tmpInt64)
+		//txnFee = uint64(tmpInt64)
```
Comment (olemis) [inline src/coin/skycoin/models/wallet.go]: @stdevMauricio1802 why is this change needed ?

Comment (mauricio1802) [reply src/coin/skycoin/models/wallet.go]: @olemis because when the incoming txn is a readableTxn the fee is not needed for create the returned txn

**Review by olemis — COMMENTED** [review]
needs control to input password e.g. for decrypting wallet

**Comment by olemis** [comment]
@stdevMauricio1802 do I have green light to review and merge ?
